### PR TITLE
feat: add logsBloom class

### LINKED ISSUE #2683: Add logsBloom class
### Problem

Currently we don't have any way to calculate logsBloom, if it's not added in the response from the mirror node.

### Solution

Add logsBloom class to handle all calculations.

### PR DESCRIPTION
**Pull request by georgi-l95**
**Description**:
This PR adds logsBloom object, which can take care of all bloom calculations, needed when there is missing logsBloom. Example are synthetic transactions, where we don't have logsBloom from the mirror node response. This PR only adds the class for this, calculations will be done in another to keep them tight and focused. 

**Related issue(s)**:

Fixes #2683 

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: packages/relay/src/lib/utils/bloom/index.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import { keccak256 } from 'ethereum-cryptography/keccak.js';
+
+const BYTE_SIZE = 256;
```
Comment (quiet-node) [inline packages/relay/src/lib/utils/bloom/index.ts:23]: We might want to put this constant into a shared common place. I forgot what it is but there's a file we store constants, right?

File: packages/relay/src/lib/utils/bloom/index.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  bitvector: Uint8Array;
+
+  /**
+   * Represents a Bloom filter.
+   */
+  constructor(bitvector?: Uint8Array) {
+    if (!bitvector) {
+      this.bitvector = new Uint8Array(BYTE_SIZE);
+    } else {
+      if (bitvector.length !== BYTE_SIZE) throw new Error('bitvectors must be 2048 bits long');
```
Comment (quiet-node) [inline packages/relay/src/lib/utils/bloom/index.ts:35]: Would be nice to have a refference link states why logsBloom must be 2048 bits (256 bytes).

File: packages/relay/src/lib/utils/bloom/index.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  /**
+   * Adds an element to a bit vector of a 64 byte bloom filter.
+   * @param e - The element to add
+   */
+  add(e: Uint8Array) {
+    e = keccak256(e);
+    const mask = 2047; // binary 11111111111
+
+    for (let i = 0; i < 3; i++) {
+      const first2bytes = new DataView(e.buffer).getUint16(i * 2);
```
Comment (quiet-node) [inline packages/relay/src/lib/utils/bloom/index.ts:49]: Would be nice to explain along the steps through code comments

File: packages/relay/src/lib/utils/bloom/index.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+  /**
+   * Adds an element to a bit vector of a 64 byte bloom filter.
+   * @param e - The element to add
+   */
+  add(e: Uint8Array) {
+    e = keccak256(e);
+    const mask = 2047; // binary 11111111111
+
+    for (let i = 0; i < 3; i++) {
```
Comment (quiet-node) [inline packages/relay/src/lib/utils/bloom/index.ts:48]: Why 3?

Comment (quiet-node) [reply packages/relay/src/lib/utils/bloom/index.ts:48]: Would be nice to explain along the steps through code comments

File: packages/relay/src/lib/utils/bloom/index.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+  /**
+   * Checks if an element is in the bloom.
+   * @param e - The element to check
+   */
+  check(e: Uint8Array): boolean {
+    e = keccak256(e);
+    const mask = 2047; // binary 11111111111
+    let match = true;
+
+    for (let i = 0; i < 3 && match; i++) {
```
Comment (quiet-node) [inline packages/relay/src/lib/utils/bloom/index.ts:66]: Would be nice to explain along the steps through code comments

**Review by quiet-node — COMMENTED** [review]
Great work just some Qs around

**Review by Nana-EC — COMMENTED** [review]
Nice work unblocking this.
I think @quiet-node had some good requests in here.
This is one of those classes that is likely rarely going to be updated but his deep crypto logic that is good to make sure if well commented and referenced so that the 1 time we do have to update it we understand what's going on

**Comment by georgi-l95** [comment]
Superseded by https://github.com/hashgraph/hedera-json-rpc-relay/pull/2785

**Pull request closed without merging**